boltdesignsystem/bolt
# [6.x] DS-672: Upgrade to webpack 5

## Jira

https://pegadigitalit.atlassian.net/browse/DS-672

## Summary

Upgrade Webpack to v5, cleanup old and unused code, update browserslist config.

## Details

- Upgrade from Webpack v4 to v5 in both Bolt build and PL build
- Update Webpack configuration
- Update dependencies for compatibility
- Fix CSS throwing errors in upgraded SCSS loader
- Resolve dependency mismatch warnings
- Update browserslist to match our actual browser support strategy
- Remove unused dependencies
- Remove unused packages:
  - packages/website-ui/uikit-critical-css-webpack-plugin/**
  - packages/website-ui/uikit-prerenderer/**
  - packages/website-ui/uikit-prerender-spa-plugin/**
- Remove unused features:
  - criticalcss
- Remove dead build-related code

## How to test

- Review code
- Verify build, watch, test commands all work as expected
- Review a variety of pages in PL for regressions

## Release notes
- Webpack in `@bolt/build-tools` has been upgraded from v4 to v5.
- Browserslist configuration in `@bolt/browserslist-config` has been updated.
